version/distro/distro.go

```diff
@@ -93,8 +101,56 @@ func linuxDistro() Distro {
 		return WDMyCloud
 	case have("/etc/unraid-version"):
 		return Unraid
+	default:
+		if d, err := parseLinuxOSRelease("/etc/os-release"); err == nil {
+			return d
+		}
+		return Unknown
 	}
-	return ""
+}
+
+// parseLinuxOSRelease parses an os-release file from path (usually
+// /etc/os-release) to extract the distro ID. If ID_LIKE is present, it is
+// prioritized over ID.
+//
+// See https://www.freedesktop.org/software/systemd/man/os-release.html
+func parseLinuxOSRelease(path string) (Distro, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return Unknown, err
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+	var id, idLike string
+	for s.Scan() {
+		line := mem.S(s.Text())
+		if mem.HasPrefix(line, mem.S("ID=")) {
+			id = mem.TrimPrefix(line, mem.S("ID=")).StringCopy()
+			id = strings.Trim(id, `"'`)
+		}
+		if mem.HasPrefix(line, mem.S("ID_LIKE=")) {
+			idLike = mem.TrimPrefix(line, mem.S("ID_LIKE=")).StringCopy()
+			idLike = strings.Trim(idLike, `"'`)
+			// ID_LIKE value can be a space-separated list of IDs. Take the
+			// last mentioned ID, which for most distros seems to be the oldest
+			// in the lineage.
+			idLikeItems := strings.Fields(idLike)
+			if len(idLikeItems) == 0 {
+				continue
+			}
+			idLike = idLikeItems[len(idLikeItems)-1]
+		}
+	}
+	if err := s.Err(); err != nil {
+		return Unknown, err
+	}
+	if idLike != "" {
```

Comment:
should we have a denylist of values we never want an idLike value of? such as "ubuntu" for now?

Reply:
Added a denylist on the return value of `parseLinuxOSRelease`, whether it's coming from `ID` or `ID_LIKE`.